generator/generator.go

```diff
@@ -408,21 +409,53 @@ func (g *generator) Build(
 		}
 	}
 
-	if !hasPointerStructMethod && ((source.Named && !source.Basic) || (target.Named && !target.Basic) || (source.Pointer && source.PointerInner.Named && !source.PointerInner.Basic)) {
+	isSourceNamed, isTargetNamed := source.Named && !source.Basic, target.Named && !target.Basic
+	isSourcePointerNamed := source.Pointer && source.PointerInner.Named && !source.PointerInner.Basic
+	isTargetPointerNamed := target.Pointer && target.PointerInner.Named && !target.PointerInner.Basic
+	if !hasPointerStructMethod && (isSourceNamed || isTargetNamed || isSourcePointerNamed || isTargetPointerNamed) {
 		name := g.namer.Name(source.UnescapedID() + "To" + strings.Title(target.UnescapedID()))
 
+		// Prepare method fields using context fields as a base.
+		methodFields := map[string]*builder.FieldMapping{}
+		for fieldName, fieldValue := range ctx.Fields {
+			// Copy all context fields but skip ignored fields.
+			// This adds all fields to the possibly reused method.
+			// See `struct_pointer_nested_ignore.yml` test case.
+			if !fieldValue.Ignore {
+				methodFields[fieldName] = fieldValue
+			}
+		}
+
+		// Prepare method flags using context flags as a base.
+		// Use only certain flag types from the context flags.
+		methodFlags := builder.ConversionFlags{}
+		for flagType, flagStatus := range ctx.Flags {
+			switch flagType {
+			case builder.FlagWrapErrors:
+				methodFlags[flagType] = flagStatus
+			case builder.FlagMatchIgnoreCase:
+				methodFlags[flagType] = flagStatus
+			case builder.FlagIgnoreMissing:
+				// See `ignore_missing_no_propagate.yml` test case.
+			case builder.FlagIgnoreUnexported:
+				// See `ignore_missing_no_propagate.yml` test case.
+			}
+		}
+		for flagType, flagStatus := range g.flags {
+			methodFlags[flagType] = flagStatus
+		}
+
 		method := &methodDefinition{
 			ID:     name,
 			Name:   name,
 			Source: xtype.TypeOf(source.T),
 			Target: xtype.TypeOf(target.T),
-			Fields: map[string]*builder.FieldMapping{},
-			Flags:  g.flags,
+			Fields: methodFields,
+			Flags:  methodFlags,
 			Call:   jen.Id(xtype.ThisVar).Dot(name),
 		}
 
 		g.lookup[signature] = method
-		g.namer.Register(method.Name)
```

Comment:
The `namer.Register()` is already called via `g.namer.Name(...)` several lines above.